src/custom/state/orders/updaters/PendingOrdersUpdater.ts

```diff
@@ -117,14 +117,22 @@ async function _updateOrders({
   const pending = orders.filter(({ owner }) => owner.toLowerCase() === lowerCaseAccount)
 
   // Exit early when there are no pending orders
-  if (pending.length === 0) {
-    // console.debug('[PendingOrdersUpdater] No orders to update')
+  if (!pending.length) {
     return
-  } /* else {
-    console.debug(
-      `[PendingOrdersUpdater] Update ${pending.length} orders for account ${account} and network ${chainId}`
-    )
-  }*/
+  } else {
+    // Check if there is any order pending for a long time
+    // If so, trigger appzi
+    const now = Date.now()
+    for (const { openSince } of pending) {
+      // Check if there's any pending for more than `PENDING_TOO_LONG_TIME`
+      if (openSince && now - openSince > PENDING_TOO_LONG_TIME) {
+        // Trigger NPS display, controlled by Appzi
+        openNpsAppziSometimes()
```

Comment:
Every time we try to update pending orders, check if any of them has been pending for more than X time.
Trigger appzi if more than that

Reply:
It doesn't need to be included in this version, but could you fo a follow up PR to also trigger a Google Analytics event for this case?

Reply:
One issue that I have regarding this is that, once the order is pending for more than X time, it'll always trigger.
Not a problem for Appzi as the display is controlled on their side, but for analytics it'll create a new event every time the update is triggered, which I believe is every 5sec

Of course we can track whether the order has been ackowledged, but that's yet another flag to the global state.

Any better idea?